src/Integration.Vsix/CFamily/MessageHandler.cs

```diff
@@ -105,7 +105,10 @@ public void HandleMessage(Message message)
                     logger.WriteLine(CFamilyStrings.MsgHandler_ReportUnexpectedFailure, message.Text);
                     break;
 
-                case "internal.fileDependency": // not currently handled. See https://github.com/SonarSource/sonarlint-visualstudio/issues/2611
+                // Rules that start with internal shouldn't be treated as issues.
+                // Some of them should be handled like `internal.fileDependency`. See: https://github.com/SonarSource/sonarlint-visualstudio/issues/2611
+                // Others should can simply ignored like `internal.z3RefutationRate`, which is used to log in the scanner how many issues are rejected by the Z3 solver 
```

Comment:
is this a bug fix? is this something that we already accidently report to the user?

Reply:
It can be after the update but before it wasn't since most of the internal messages has invalid loc (line number 0..) which is ignored.
I added this here because it is the right behavior and it impacted the integration test after fixing the bug about Z3 (#2875) in the "Use the subprocess SonarLint mode" branch.

By the way, you are reviewing the #2661 code in the update of analyzer PR. even though the dependents on each others might be better to separate the review.

Reply:
Thanks for pointing that out, this branch contained everything, hence I missed the other PR. 
I've changed the base of this branch to point to the other branch, and it makes it much simpler now.